code/datums/components/pitfall.dm

```diff
@@ -0,0 +1,199 @@
+/** There are many pitfalls in the game. Here is a list of them, for easier maintainence.
+ * Oshan/nadir trench:	1 direct drop, 1 area
+ * Polaris Pit:			2 landmarks
+ * sea elevator shaft:	1 landmark
+ * Icemoon:				2 landmarks
+ * Biodome:				3 landmarks
+ *
+ *
+ * These are ALL TURFS. They should STAY TURFS.
+ * similar but not quite the same as /datum/component/teleport_on_enter
+ *
+ * Here is a note preserved from the initial commit, referring to the icemoon abyss pitfall
+ * 	// this is the code for falling from abyss into ice caves
+ *	// could maybe use an animation, or better text. perhaps a slide whistle ogg?
+ **/
+
+ABSTRACT_TYPE(/datum/component/pitfall)
+/// A component for turfs which make movable atoms "fall down a pit"
+/datum/component/pitfall
+	/// a list of targets for the fall to pick from
+	var/list/TargetList = list()
```

Comment:
This too doesn't need to exist on the parent imo. It is not necessary to really have this for the landmark fall and arguably also not for the area fall. (To keep the current behaviour of z level fall it does need to exist I think but also the current behaviour of z level falls is kinda weird. Changing it is out of the scope of this PR though.)